components/director/internal/domain/oauth20/service.go

```diff
@@ -51,7 +53,9 @@ func NewService(scopeCfgProvider ScopeCfgProvider, uidService UIDService, cfg Co
 func (s *service) CreateClientCredentials(ctx context.Context, objectType model.SystemAuthReferenceObjectType) (*model.OAuthCredentialDataInput, error) {
 	scopes, err := s.getClientCredentialScopes(objectType)
 	if err != nil {
-		return nil, err
+		if !(apperrors.IsEmptyTenant(err) && objectType == model.IntegrationSystemReference) {
```

Comment:
Can you refactor this if-statement to be more descriptive? Maybe just extract this part to some simple function that will return a bool and named it to reveal the actual intention. I also see similar patter in multiple lines so please align the other places.